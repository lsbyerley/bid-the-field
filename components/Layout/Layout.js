import NavBar from '../NavBar';

const Layout = ({ children }) => {
  return (
    <div className='flex flex-col min-h-screen grid-bg'>
      <NavBar />
      <main className='flex-1'>{children}</main>
      <footer className='flex items-center justify-center py-8 bg-base-200'>
        <a
          href='https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app'
          target='_blank'
          rel='noopener noreferrer'
        >
          Bid The Field{' '}
        </a>
      </footer>
    </div>
  );
};

export default Layout;
