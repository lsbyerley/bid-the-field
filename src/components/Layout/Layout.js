import Link from 'next/link';
import NavBar from '../NavBar';

const Layout = ({ children }) => {
  return (
    <div className='flex flex-col min-h-screen bg-base-200 grid-bg'>
      <NavBar />
      <main className='flex-1'>{children}</main>
      <footer className='flex items-center justify-center py-8 bg-base-100'>
        <Link href='/'>
          <a>Bid The Field</a>
        </Link>
      </footer>
    </div>
  );
};

export default Layout;
