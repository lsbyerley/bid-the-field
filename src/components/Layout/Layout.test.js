import React from 'react';
import { render, screen } from 'test-utils';
import Layout from './Layout';

jest.mock('../NavBar', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation((props) => {
    return <div role='nav'>{JSON.stringify(props)}</div>;
  }),
}));

const renderLayout = (props) => {
  const utils = render(<Layout {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('Layout', () => {
  it('should render and match snapshot', () => {
    const { home } = renderLayout();
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
