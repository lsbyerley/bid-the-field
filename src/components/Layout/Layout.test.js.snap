// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render and match snapshot 1`] = `
<div
  class="flex flex-col min-h-screen bg-base-200 grid-bg"
>
  <div
    role="nav"
  >
    {}
  </div>
  <main
    class="flex-1"
  />
  <footer
    class="flex items-center justify-center py-8 bg-base-100"
  >
    <a
      href="/"
    >
      Bid The Field
    </a>
  </footer>
</div>
`;
